src/shared/components/query/QueryStore.ts

```diff
@@ -143,46 +145,35 @@ export class QueryStore
 	}
 
 	@observable studiesHaveChangedSinceInitialization:boolean = false;
-	@observable savedVirtualCohorts:VirtualCohort[] = [];
+	@observable userVirtualStudies:VirtualStudy[] = [];
+	@observable deletedVirtualStudies:string[] = [];
 
 	@action public deleteVirtualCohort(id:string) {
-		VirtualCohorts.delete(id);
+		sessionServiceClient.deleteVirtualStudy(id)

```

Comment:
i think it's antipattern to tag an async method with action.   instead action should just suround the code which actually mutation state.

Reply:
see mobx action documentation for how to do that.